### PR TITLE
WSTEAM1-355: Create media promo list and promo list item component

### PR DESCRIPTION
**Pull request by Isabella-Mitchell**
Resolves JIRA WSTEAM1-355 and WSTEAM1-353 (subtasks of WSTEAM1-307)

Overall changes
======
Create a new 'Latest Media' component to show on Media Article Pages.

Code changes
======
- Create a `LatestMediaSection` within `MediaArticlePage` > `PagePromoSections`.
- Update the MediaArticlePage SecondaryColumn file to show the Latest Media component.
- Add Fixture data for Media Article Pages (Hausa, Pidgin, Tamil)
- Add unit tests
- Add stories for the LatestMediaSection and update Media Article Page stories.

Outstanding
======
- Remove Top Stories from `MediaArticlePage` > `PagePromoSections`. We will do this once the PR is code reviewed.

Testing
======
- Pull down branch and launch the local server
- Visit the following links to to see the new component
- http://localhost:7080/hausa/articles/cxr0765kxlzo
- http://localhost:7080/tamil/articles/c84m2jl4dpzo
- http://localhost:7080/pidgin/articles/cw0x29n2pvqo

To view without firing up a local dev server, use our Storybook stories:
- https://wsteam1-355-media-promo-list-componen--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=pages-media-article-page--media-article-page&viewMode=story
- https://wsteam1-355-media-promo-list-componen--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=pages-media-article-page--media-article-page-with-latest-media-images&viewMode=story
- https://wsteam1-355-media-promo-list-componen--5d28eb3fe163f6002046d6fa.chromatic.com/iframe.html?args=&id=pages-media-article-page--media-article-page-with-single-latest-media&viewMode=story

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaIndicator/index.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaIndicator/index.tsx]: Could we use `Promo.MediaIndicator` here and pass in the css using the CSS prop, like `css={styles.promoMediaIndicator}`. I think that would result in the same output.

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaIndicator/index.styles.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaIndicator/index.styles.tsx]: This can be written in the shorthand `padding: '0.5rem 0.125rem'`.

In this format, the first value applies to the `top | bottom` attributes and the second value applies to the `left | right` attributes.

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaItem/index.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaItem/index.tsx]: Similar to the other comment, I think we can import the `Promo` component directly here and use the `css` prop to pass the styles, rather than styling it in the CSS file and then importing as a Styled component.

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx]: Similar to above, this could be short handed to `margin: 0 0.5rem`, although not a huge deal as it may become less readable?

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx]: Wee typo here: `marginBottom`

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: Similar to above, is it possible to use the `Promo` component directly and pass the css via the `css` prop?

File: src/app/pages/MediaArticlePage/SecondaryColumn.jsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/SecondaryColumn.jsx]: Could this styling be moved into the `styles.ts` file? Would save having to import the GEL spacings and breakpoints as they are available on the Theme.

**Comment by amoore108** [comment]
Just a few comments around the usage of Styled components, but it looks good!

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaIndicator/index.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaIndicator/index.tsx]: I think we did this originally because the styles we were applying weren't being applied due to the mixture of css prop and the styled api in OptimoPromos. In the end we had to add classNames anyway because styles weren't behaving properly so this should work now

File: src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts
Comment (amoore108) [inline src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts]: These `GEL_` spacings should have their equivalents available on `spacings` from the Theme, so we could use those and remove the Psammead import.

**Comment by amoore108** [comment]
Thanks for making those changes, looks good! 

Just one thing I noticed, the component folder is called `LastestMedia` 😅

File: src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts]: And just to add to what Aaron has mentioned above, if you do use the spacings from the Theme, you need to append `rem` to the end 

e.g. 
```suggestion
  responsiveComponentWrapper: ({ mq, spacings }: Theme) =>
    css({
      marginBottom: `${spacings.TRIPLE}rem`,
      [mq.GROUP_4_MIN_WIDTH]: {
        marginBottom: `${spacings.FULL}rem`,
        padding: `${spacings.DOUBLE}rem`,
      },
```

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx]: Have we defined this as a type somewhere (perhaps in globals types) and we could reuse? If not, is it worth creating it?

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.stories.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.stories.tsx]: Not sure if this is relevant / possible, but all stories now include almost all of the providers - https://github.com/bbc/simorgh/blob/latest/.storybook/preview.js#L557-L570 - which was merged in  #10699 - wonder if you can remove some of the providers from here?

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx]: **nitpick**: Should the style names start with lowercase - given that they are just fields on an object? This is also inconsistent with some of the other styles we have defined as part of this work e.g. in https://github.com/bbc/simorgh/blob/WSTEAM1-355-media-promo-list-component/src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaItem/index.styles.tsx the style names begin with lowercase.
```suggestion
  latestMediaSection: ({ mq, isDarkUi, palette }: Theme) =>
```

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: **nitpick**: Does this read better?
```suggestion
  const hasSingleItem = content.length === 1;
```

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: I am seeing a wee bit of duplication between this and within `renderLatestMediaList` above. Wondering if there might be a nicer way of consolidating this? i.e. instead of `renderLatestMediaList` could it just be `renderLatestMedia` and you pass a single latest media item to it?

Then you could reuse it in both places?

Alternatively, my preference (but maybe get a second opinion on this) would be to invoke the component code directly rather than abstracting it within a function - makes it much easier to read and understand what a component is doing. 

Will demonstrate 👇

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: Per my comment above - you may need to work out a way of calculating the aria label for each entry, but would be better to see if here, rather than abstracted into a function.

```suggestion
            <PromoItem key={index} css={styles.LatestMediaPromoBorderAndWidth}>
              <LatestMediaItem
                  item={item}
                  ariaLabelledBy={ariaLabelledBy}
                  ref={viewRef}
                  eventTrackingData={eventTrackingData}
               />
           </PromoItem>
```

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx:1]: Is it worth just renaming this to `types.ts` (it doesn't need to be a .tsx file) and having it at the same level as the `LatestMediaSection` directory. I think we have done this in a few other places.

**Review by karinathomasbbc — COMMENTED** [review]
This looks great, sorry about all of my comments - not all of them are important, so just have a look and see what you think. Happy to talk through anything that's not clear.

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx]: I guess this depends if we're looking to use this data structure elsewhere in the application. From what I know this is the only place it's going to be used, so not sure if it's worth making it a global type and just keeping it co-located?

Comment (karinathomasbbc) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx]: Yeah that's fair - why don't we just add a new local type in this file and reference it here? We can always extract / refactor / reuse it later. Thanks.

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/LatestMediaItem/index.test.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/LatestMediaItem/index.test.tsx]: We could import this type from the types file per my comment above.

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx
Comment (karinathomasbbc) [inline src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx]: This should start with a lower case, for consistency with other styles. 
```suggestion
  latestMediaSection: ({ mq, isDarkUi, palette }: Theme) =>
```

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx:1]: Done

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: Done

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.styles.tsx]: Done

File: src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts]: Done

Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts]: Done

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: Fully agree with this. However, this method is used in other places [here](https://github.com/bbc/simorgh/blob/6b4e1ca6e949d26702e26a435a5acab655e4adef/src/app/pages/ArticlePage/PagePromoSections/RelatedContentSection/index.jsx#L33) and [here](https://github.com/bbc/simorgh/blob/6b4e1ca6e949d26702e26a435a5acab655e4adef/src/app/pages/ArticlePage/PagePromoSections/TopStoriesSection/index.jsx#L19). So maybe we scope that work in a separate ticket to tidy up this logic to not use a render function? What do you think?

Comment (karinathomasbbc) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.tsx]: Yes, a new ticket to tidy up / refactor works for me. If we could manage to do it not too long after merging this code, that would be best - it's more likely to be forgotten about if we leave it.

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.stories.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.stories.tsx]: Now that I'm looking at the the global provider method, it looks a little complicated then I first thought, there's a bit of overhead with [updating the global values from within a story](https://storybook.js.org/docs/react/essentials/toolbars-and-globals#updating-globals-from-within-an-addon). I'm going to speak with Dario about it, so for now I think we leave it like this if that's okay?

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx]: Sorry I think I got a bit confused, I thought you were referring to the LatestMedia type as a whole, are you referring to those four keys as being a type like this
```ts
type Example = {
  imageUrl: string;
  link: string;
  title: string;
  type: 'video' | 'audio';
};
```
If so what name would you suggest? MediaPromo?

File: src/app/pages/MediaArticlePage/SecondaryColumn.jsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/SecondaryColumn.jsx]: These could be removed along with the check on line:21 as they aren't needed anymore.

**Review by amoore108 — APPROVED** [review]
Looks great! Just a removal of some now unneeded lines.

File: src/app/pages/MediaArticlePage/SecondaryColumn.jsx
Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/SecondaryColumn.jsx]: Done

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.stories.tsx
Comment (karinathomasbbc) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/index.stories.tsx]: That's absolutely fine, I didn't expect you to make the changes as part of this, or only if it was very simple and reduced duplication. Definitely worth knowing for the future if we need to add all the providers to our storybook stories or not (I am also keen to know how this should work going forward!)

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx
Comment (karinathomasbbc) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx]: I was actually refering to the `type: 'video' | 'audio';` line (the GitHub conversations UI is a bit confusing - I prefer the files view!)

```
export type Media = 'video' | 'audio';
```

And then where it's declared
```
...
type: Media;
```

Comment (HarveyPeachey) [reply src/app/pages/MediaArticlePage/PagePromoSections/LastestMediaSection/LatestMediaTypes/index.tsx]: Ahh yeah, makes sense, I've just added it!

**Comment by greenc05** [comment]
Can we add the Health Factors component to this please? https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/docs/components-healthfactors-healthfactorscomponent--health-factors-with-complete-docs 

And also link the Figma design files in storybook if that's not done? https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/docs/coding-standards-storybook--page#how-to-add-figma-previews

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx
Comment (amoore108) [inline src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx:8]: Just checked with @emilymynall, this margin should only apply on the mobile breakpoint when it goes back to a single column of promo items. On desktop the margin can be removed and I think the tablet applies its own `margin: 0 1rem` anyway.

File: src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts
Comment (amoore108) [inline src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts]: Just checked with @emilymynall, this padding can be removed. The promos should take up the entire width available of the SecondaryColumn column.

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx
Comment (greenc05) [inline src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx]: Not sure you need 'rem'? 0 is 0 whatever the unit?

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx
Comment (Isabella-Mitchell) [reply src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx:8]: I have resolved by adding a [mq.GROUP_4_MIN_WIDTH]: media query to remove the margin. I decided to do this as it follows 'mobile' first approach. 

An alternative would be to apply this margin only to 0/1/2 breakpoints, and then make use of the tablet auto margin.

Let me know if you think I should update it.

File: src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts
Comment (Isabella-Mitchell) [reply src/app/pages/MediaArticlePage/MediaArticlePage.styles.ts]: 👍

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx
Comment (Isabella-Mitchell) [reply src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx]: good spot! Have fixed

File: src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx
Comment (amoore108) [reply src/app/pages/MediaArticlePage/PagePromoSections/LatestMediaSection/index.styles.tsx:8]: What you have now looks good to me!

**Review by amoore108 — APPROVED** [review]

**Review by amoore108 — APPROVED** [review]

**Pull request merged**
